### PR TITLE
fix(rpc): better error messages on RPC EOF

### PR DESCRIPTION
**Pull request by Arqu**
https://github.com/qri-io/qri/issues/1227

### CODE REVIEW

File: lib/update.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func (m *UpdateMethods) Schedule(in *ScheduleParams, out *cron.Job) (err error)
 		return fmt.Errorf("apologies, we currently can't schedule updates while `qri connect` is running")
 		// TODO (b5) - this returns error EOF when run. need to figure out why. I suspect it's b/c
 		// job contains an interface field: Options
-		// return m.inst.rpc.Call("UpdateMethods.Schedule", in, out)
+		// return checkRPCError(m.inst.rpc.Call("UpdateMethods.Schedule", in, out))
```
Comment (Arqu) [inline lib/update.go]: Will investigate this as part of https://github.com/qri-io/qri/issues/813

File: lib/lib.go [hunk 2]
```diff
@@ -743,3 +744,20 @@ func (inst *Instance) RemoteClient() remote.Client {
 func (inst *Instance) Teardown() {
 	inst.teardown()
 }
+
+// checkRPCError validates RPC errors and in case of EOF returns a
+// more user friendly message
+func checkRPCError(err error) error {
+	if strings.Contains(err.Error(), "EOF") {
```
Comment (b5) [inline lib/lib.go]: err will often be `nil` here when it works, and will panic when the `Error()` method is called on a nil value. Adding this above will fix:

```go
if err == nil {
  return nil
}
```

File: lib/lib.go [hunk 3]
```diff
@@ -743,3 +744,20 @@ func (inst *Instance) RemoteClient() remote.Client {
...
+		msg := `Qri wasn't able to execute your command. 
+It is possible you have a Qri node running or the Desktop app is open.
+Try closing them and running the command again.
+Check our issue tracker for RPC issues & feature requests:
+  https://github.com/qri-io/qri/issues?q=is:issue+label:RPC
+
+Error:
+%s`
```
Comment (b5) [inline lib/lib.go]: Language of this looks great. 

I just kinda hacked together this convention in the SQL pr, does it work for you? More info we could include?

Comment (Arqu) [reply lib/lib.go]: Yeah I saw that back in the PR and thought it was a bit ellaborate but very much user friendly. 
Just a couple bits on the language itself:
Qri actually does execute the command but couldn't parse the response (and we don't know if it was successful or not)! maybe should change to this.
Other than that, I'd just like more "specific" error handling and not just a blanket RPC error. Though that can come over time.

Comment (b5) [reply lib/lib.go]: totally. At least this gives clear feedback. We can workshop the friendly error format as time goes on 😄

File: lib/lib.go (hunk 2 above)
Comment (b5) [reply lib/lib.go]: very sad to see this didn't crash the test suite anywhere.

Comment (Arqu) [reply lib/lib.go]: Yeah, thing is I thought adding a whole test suite for the RPCs against a running qri node. But thats quite a substantial bit of work and not really sure if it's worth the time if we're going to move away from it anyways.

Comment (b5) [reply lib/lib.go]: completely agreed. I'd _much_ rather see that energy go into refactoring our api package tests

**Pull request merged**